src/ConcordiumNetSdk/Types/AccountNonce.cs

```diff
@@ -0,0 +1,88 @@
+﻿using ConcordiumNetSdk.Helpers;
+
+namespace ConcordiumNetSdk.Types;
+
+/// <summary>
+/// Represents an account sequence number.
+///
+/// The account sequence number is a 64-bit unsigned integer specific to an account
+/// and used when submitting account transactions for that account to the node. The account
+/// sequence number is maintained as on-chain state and is incremented with each finalized
+/// transaction. The next sequence number to be used in a transaction can be queried
+/// with <see cref="Client.GetNextAccountSequenceNumber"> or
+/// <see cref="Client.GetAccountInfo"/>.
+/// </summary>
+public readonly struct AccountNonce : IEquatable<AccountNonce>
+{
+    public const UInt32 BytesLength = sizeof(UInt64);
+
+    /// <summary>
+    /// The value of the nonce.
+    /// </summary>
+    public readonly UInt64 Value { get; init; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AccountNonce"/> class.
+    /// </summary>
+    /// <param name="nonce">The nonce.</param>
+    public AccountNonce(UInt64 nonce)
+    {
+        Value = nonce;
+    }
```

Comment:
Nonce is >= 1. So that validation should be done.

Also we started calling this "SequenceNumber" in V2 API. So I suggest using that name instead of Nonce.